homeassistant/components/heos/media_player.py

```diff
@@ -128,6 +136,65 @@ async def async_clear_playlist(self):
         """Clear players playlist."""
         await self._player.clear_queue()
 
+    async def async_get_group(self):
+        """Build a dictionary with HEOS group membership information."""
+        group_membership = []
+        if not self.player_id:
+            _LOGGER.info(
+                "No player_id for %s yet, not returning group info", self.entity_id
+            )
+            return group_membership
+
+        controller = self.hass.data[HEOS_DOMAIN][DATA_CONTROLLER_MANAGER].controller
+        try:
+            groups = await controller.get_groups(refresh=True)
+        except HeosError as err:
+            _LOGGER.error("HEOS Unable to get group info: %s", err)
+            return group_membership
+
+        for group in groups.values():
+            member_player_ids = {member.player_id for member in group.members}
+            if (
+                self.player_id == group.leader.player_id
+                or self.player_id in member_player_ids
+            ):
+                leader_entity_id = None
+
+                for entity in self.hass.data[DOMAIN].entities:
+                    if not isinstance(entity, HeosMediaPlayer):
+                        continue
+                    if entity.player_id == group.leader.player_id:
+                        leader_entity_id = entity.entity_id
+                    elif entity.player_id in member_player_ids:
+                        group_membership.append(entity.entity_id)
+                # Make sure the group leader is always the first element
+                group_membership.insert(0, leader_entity_id)
+                return group_membership
+        return group_membership
+
+    @log_command_error("join_players")
+    async def async_join_players(self, group_members):
+        """Join `group_members` as a player group with the current player."""
+        player_ids = []
+        for player_entity_id in group_members:
+            for entity in self.hass.data[DOMAIN].entities:
+                if (
+                    isinstance(entity, HeosMediaPlayer)
+                    and entity.entity_id == player_entity_id
+                ):
+                    player_ids.append(entity.player_id)
+
+        controller = self.hass.data[HEOS_DOMAIN][DATA_CONTROLLER_MANAGER].controller
+        try:
+            await controller.create_group(self.player_id, player_ids)
+        except HeosError as err:
+            _LOGGER.error(
+                "HEOS create_group with arguments (%s %s) failed. Error: %s",
```

Comment:
Don't leak the underlying API and parameters into user-facing error messages. Instead, log something that is understandable to a user and reference the entity IDs: 
```suggestion
                "Failed to group %s with %s: %s",
```